t/plugin/basic-auth.t

```diff
@@ -356,3 +356,55 @@ GET /t
 GET /t
 --- no_error_log
 [error]
+
+
+
+=== TEST 15: enable basic auth plugin using admin api, set hide_credentials = true
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "basic-auth": {
+                            "hide_credentials": true
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: verify
```

Comment:
Can we strengthen this case that we can make sure the header is really removed, rather than just checking out the log.

Reply:
I am ashamed not found such parse in project of test-nginx, how could we detect request header?

Reply:
Try `response_headers` ?

Reply:
The `response_headers` data section can be used to validate response header entries, but `Authorization` is request header, so I don't know whether `response_headers` can used in this test case.

Reply:
we can mock an upstream, and check headers in upstream, take a look at: https://github.com/apache/apisix/blob/70174d58cd4248f55dada5c2bc7396cdf73972c7/t/plugin/proxy-mirror.t#L35-L55

Reply:
Done. please see cb53193